### PR TITLE
feat: handle route for a project which has no site

### PR DESCRIPTION
**Pull request by sunilsabatp**
- [ ] Handle case of setting site id to null when project has no sites. example (mothersforest project unselecting a plant location; ploc param remains in the url) (visit to -> `en/prd/mothersforest`)

### CODE REVIEW

**Review by mariahosfeld — COMMENTED** [review]
Functionality looks good to me.

File: src/features/projectsV2/ProjectsMap/index.tsx [hunk 1]
```diff
@@ -92,22 +92,28 @@ function ProjectsMap(props: ProjectsMapProps) {
   const onClick = useCallback(
     (e) => {
       if (props.page !== 'project-details') return;
+      const hasNoSites =
+        singleProject?.sites && singleProject?.sites?.length === 0;
       const result = getPlantLocationInfo(plantLocations, mapRef, e.point);
```
Comment (mohitb35) [inline src/features/projectsV2/ProjectsMap/index.tsx:97]: You could probably shorten this condition to 

`singleProject?.sites?.length === 0`

File: src/features/projectsV2/ProjectsContext.tsx [hunk 2]
```diff
@@ -286,17 +286,22 @@ export const ProjectsProvider: FC<ProjectsProviderProps> = ({
...
-      const result = plantLocations?.find(
-        (plantLocation) => plantLocation.hid === requestedPlantLocation
-      );
-      if (result) {
-        setSelectedPlantLocation(result);
+      const hasNoSites = singleProject.sites?.length === 0;
+
+      if (hasNoSites) {
+        //Case when a direct link requests a specific plant location but no sites exist for a project(e.g projectSlug: mothersforest).
+        updateUrlWithSiteId(locale, singleProject.slug, null);
```
Comment (mohitb35) [inline src/features/projectsV2/ProjectsContext.tsx:294]: This logic is a little confusing to me.

Why are we calling `updateUrlWithSiteId` directly? Why not pass in `null` to `updateSiteAndUrl`?

**Review by mohitb35 — CHANGES_REQUESTED** [review]
@sunilsabatp Please go through my comments

**Review by mohitb35 — APPROVED** [review]
LGTM

**Pull request merged**